ui/hooks/metamask-notifications/useProfileSyncing.ts

```diff
@@ -83,6 +85,9 @@ export function useDisableProfileSyncing(): {
     try {
       // disable profile syncing
       await dispatch(disableProfileSyncingAction());
+
+      // list notifications to update the counter
+      await listNotifications();
```

Comment:
Updating the list after the notifications have been disabled, the counter is set to zero

Reply:
nice, we can ticket this for a core change in future.